### PR TITLE
Added third party dependencies and BUILD/WORKSPACE files for a plugin

### PR DESCRIPTION
**Pull request by margaritiko**
Create a project and connect required third party libraries

* Following [Bazel best practices](https://docs.bazel.build/versions/master/best-practices.html) placed folder with third party library in the directory. 
* Added **WORKSPACE** file at top level which loads `dagger`, `dagger-spi` and third party auto-services library.
* Specified java_plugin in **BUILD** file with source class `QueryPlugin`.

[Card](https://github.com/googleinterns/dagger-query/projects/1#card-42487403)

### CODE REVIEW

File: project/WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    repositories = DAGGER_REPOSITORIES,
+)
+
+load("//:workspace_defs.bzl", "google_common_workspace_rules")
+
+google_common_workspace_rules()
+
+load("@bazel_skylib//:workspace.bzl", "bazel_skylib_workspace")
+
+bazel_skylib_workspace()
```
Comment (Ratchette) [inline project/WORKSPACE]: is this currently used anywhere?

File: project/third_party/java/auto/BUILD [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        ":common",
+        ":service",
+        "//third_party/java/google_java_format",
+        "//third_party/java/guava",
+        "//third_party/java/javapoet",
+        "@com_google_auto_factory_auto_factory//jar",
+    ],
+)
+
+java_library(
```
Comment (Ratchette) [inline project/third_party/java/auto/BUILD]: Can you please remove the targets that are not currently being used?
(In all of the files of this PR)

I think it's good practice to include build dependencies in the same PR where they are first used.

File: project/src/com/google/daggerquery/QueryPlugin.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.google.daggerquery;
```
Comment (Ratchette) [inline project/src/com/google/daggerquery/QueryPlugin.java:17]: Heads up, all source code files need to have a special header that includes copyright and license information.
https://github.com/googleinterns/dagger-query#source-code-headers

**Review by Ratchette — COMMENTED** [review]
I have no idea what this button does, but it sounds important. So I'm clicking on it anyways :)

File: project/BUILD [hunk 4]
```diff
@@ -0,0 +1,11 @@
+load("@rules_java//java:defs.bzl", "java_plugin")
```
Comment (Ratchette) [inline project/BUILD:15]: Please add the copyright header here too.

File: project/third_party/java/auto/BUILD (hunk 2 above)
Comment (Ratchette) [reply project/third_party/java/auto/BUILD]: Fixed in followup commit.

File: project/WORKSPACE [hunk 5]
```diff
@@ -0,0 +1,32 @@
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
```
Comment (Ratchette) [inline project/WORKSPACE:15]: Please add a header here too just in case.

File: project/workspace_defs.bzl [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+def _maven_import(artifact, sha256, licenses, **kwargs):
+    parts = artifact.split(":")
+    group_id = parts[0]
+    artifact_id = parts[1]
+    version = parts[2]
+    name = ("%s_%s" % (group_id, artifact_id)).replace(".", "_").replace("-", "_")
+    url_suffix = "{0}/{1}/{2}/{1}-{2}.jar".format(group_id.replace(".", "/"), artifact_id, version)
+
+    # TODO(cpovirk): Consider jvm_maven_import_external.
```
Comment (Ratchette) [inline project/workspace_defs.bzl]: Can you please remove these two TODOs or update them to your Google username?

**Review by Ratchette — CHANGES_REQUESTED** [review]
Your CL is much smaller and easier to read now, thanks!

There are three last minor changes.

File: project/BUILD (hunk 4 above)
Comment (margaritiko) [reply project/BUILD:15]: Done ✅

File: project/WORKSPACE (hunk 5 above)
Comment (margaritiko) [reply project/WORKSPACE:15]: Done ✅

File: project/src/com/google/daggerquery/QueryPlugin.java (hunk 3 above)
Comment (margaritiko) [reply project/src/com/google/daggerquery/QueryPlugin.java:17]: ✅

File: project/WORKSPACE (hunk 1 above)
Comment (margaritiko) [reply project/WORKSPACE]: Hm, the last two lines are useless, I've removed them 💣

**Review by Ratchette — APPROVED** [review]
LGTM!  :poodle:
--

**Pull request merged**